pkg/client/userd/trafficmgr/intercept.go

```diff
@@ -406,12 +408,41 @@ func (tm *TrafficManager) CanIntercept(c context.Context, ir *rpc.CreateIntercep
 		return nil, interceptError(rpc.InterceptError_TRAFFIC_MANAGER_ERROR, errcat.Category(pi.ErrorCategory).Newf(pi.Error))
 	}
 
+	wl, err := k8sapi.GetWorkload(c, spec.Agent, spec.Namespace, spec.WorkloadKind)
```

Comment:
Any information needed to do the intercept must be available at this point. The client should not retrieve the workload here and reason about the `agentImage`. All of that must have been made by the traffic-manager already. If there are things missing from what's returned by `PrepareIntercept`, then it should be added to the `PreparedIntercept` gRPC message (it does have an `agentImage` field, and it will always be set).